xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -5472,6 +5472,8 @@ glusterd_compare_friend_data(dict_t *peer_data, dict_t *cmp, int32_t *status,
 out:
     if ((ret || !update) && arg) {
         dict_unref(arg->peer_data);
+        peer_data = NULL;
```

Comment:
What's supposed to do this ? `peer_data` is a local variable...

Reply:
My bad....I have updated the code.